packages/analytics-core/src/config.ts

```diff
@@ -58,6 +61,7 @@ export class Config implements IConfig {
     this.loggerProvider = options.loggerProvider || defaultConfig.loggerProvider;
     this.logLevel = options.logLevel ?? defaultConfig.logLevel;
     this.minIdLength = options.minIdLength;
+    this.plan = options.plan ?? defaultConfig.plan;
```

Comment:
no need for else case since default is `undefined` anyway

```suggestion
    this.plan = options.plan;
```